# Fix EVP_PKEY_CTX_set_rsa_oaep_md not yet implemented in LibreSSL
# This patch will probably break E2E encryption
# See: https://github.com/nextcloud/desktop/issues/738
# And: https://github.com/libressl-portable/portable/issues/547
--- src/libsync/clientsideencryption.cpp.orig
+++ src/libsync/clientsideencryption.cpp
@@ -37,2 +37,11 @@
 
+#ifndef EVP_PKEY_CTX_set_rsa_oaep_md
+#define EVP_PKEY_CTRL_RSA_OAEP_MD       (EVP_PKEY_ALG_CTRL + 9)
+#define EVP_PKEY_CTRL_GET_RSA_OAEP_MD   (EVP_PKEY_ALG_CTRL + 11)
+
+#define EVP_PKEY_CTX_set_rsa_oaep_md(ctx, md) \
+        EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_RSA, EVP_PKEY_OP_TYPE_CRYPT,  \
+                          EVP_PKEY_CTRL_RSA_OAEP_MD, 0, (void *)(md))
+#endif
+
 QDebug operator<<(QDebug out, const std::string& str)
